src/Microsoft.InsightsGenerator/Workflow.cs

```diff
@@ -12,105 +12,47 @@ namespace Microsoft.InsightsGenerator
-                        _instance.ProcessInputData(cancellationToken).ConfigureAwait(false);
-                    }
-                }
-            }
-
-            return _instance;
-        }
-
-        public void IngestSiggen(object input)
+        public async Task<string> ProcessInputData(DataArray rulesData, CancellationToken cancellationToken = new CancellationToken())
         {
-            if (_instance == null)
-            {
-                throw new ApplicationException("Please initialize the singleton class using Workflow.getInstance() and call _instance.IngestSiggen");
-            }
-
-            if (input == null)
-            {
-                throw new ArgumentException("Unable to ingest Siggen. Input object is null.");
-            }
-            _instance._siggenQueue.Enqueue(input);
-        }
-
-        public void IngestRules(DataArray input)
-        {
-            if (_instance == null)
-            {
-                throw new ApplicationException("Please initialize the singleton class using Workflow.getInstance() and call _instance.IngestRules");
-            }
-
-            if (input == null)
-            {
-                throw new ArgumentException(" Unable to ingest Rules. Input object is null.");
-            }
-            _instance._rulesQueue.Enqueue(input);
-        }
-
-        private async Task ProcessInputData(CancellationToken cancellationToken)
-        {
-
+            // added cancellationToken just in case for future
             cancellationToken.ThrowIfCancellationRequested();
 
-            //while (!cancellationToken.IsCancellationRequested)
-            //{
-            //    bool siggenInput = _instance._siggenQueue.TryDequeue(out object siggen);
-
-            //    if (siggenInput && siggen != null)
-            //    {
-            //       // call the siggen processor
-            //    }
-            //    else
-            //    {
-            //        await Task.Delay(TimeSpan.FromMinutes(1), cancellationToken).ConfigureAwait(false);
-            //    }
-            //}
+            //Get the signature result
+            SignatureGenerator siggen = new SignatureGenerator(rulesData);
 
-            bool rulesInput = _instance._rulesQueue.TryDequeue(out DataArray rulesData);
+            string insights = null;
 
-            if (rulesInput && rulesData != null)
+            await Task.Run(() =>
             {
-                //Get the signature result
-                SignatureGenerator siggen = new SignatureGenerator(rulesData);
-                SignatureGeneratorResult result = siggen.Learn();
-                // call the rules engine processor
-                if (result?.Insights == null)
+                try
                 {
-                    Console.WriteLine("Failure in generating insights, Input not recognized!");
+                    DataTransformer transformer = new DataTransformer();
+                    rulesData = transformer.Transform(rulesData);

```

Comment:
technically you don't need to assign the return code since the input is modified by reference, the return value is mostly to support fluent API calling style

Reply:
done